### PR TITLE
Replace link with button in assessment group actions dropdown

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Follow-up to https://github.com/PrairieLearn/PrairieLearn/pull/9447#discussion_r1499903176. Converts `a` tags into buttons, and moves the popover data into the button itself instead of a separate div.

Also includes controls to avoid having the popover dismiss the dropdown, and to force the popovers to hide when the dropdown is closed, based on similar controls done in #8889.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorAssessmentGroups/formDeleteGroup.ejs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <form name="delete-group-form" method="POST">
+  <p>Are you sure you want to delete the group <strong><%= row.name %></strong>?</p>
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/pages/instructorAssessmentGroups/formDeleteGroup.ejs:2]: This popover looked weird without at least some text, so I added a short confirmation text.

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs [hunk 2]
```diff
@@ -240,64 +248,55 @@
...
-                        data-html="true"
-                        data-placement="auto"
-                        title="Confirm delete this group"
-                        data-content="<%= include('formDeleteGroup.ejs', {row: row, iRow: iRow }); %>">
-                  </div>
-
-                  <%# Capture all clicks to dropdown items to prevent scrolling to the top of the page %>
-                  <div class="dropdown-menu" onclick="window.event.preventDefault();">
-
-                    <% if (authz_data.has_course_instance_permission_edit) { %>
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs:285]: This test was unnecessary, there is another similar test in line 249.

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs [hunk 3]
```diff
@@ -11,7 +11,15 @@
...
-      $('[data-toggle="popover"]').popover({ sanitize: false })
+      $('[data-toggle="popover"]').popover({ sanitize: false });
+      // Prevent the dropdown menu from closing when the popover is opened.
+      $('.js-group-action[data-toggle="popover"]').on("click", (e) => {
+        e.stopPropagation();
+      });
+      // Hide all popovers when the dropdown menu is closed.
+      $('.js-group-action-dropdown').on('hide.bs.dropdown', () => {
+        $('.js-group-action[data-toggle="popover"]').popover("hide");
+      });
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs:22]: These changes seem to have broken the "Add members" option. Clicking to focus in the UID input closes both the dropdown and the popover.

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs [hunk 4]
```diff
@@ -240,64 +248,55 @@
...
-                    <a class="dropdown-item" onclick="$('#row<%= iRow %>PopoverAdd').popover('show')" href="#">
+                  <div class="dropdown-menu">
+                    <button
+                      id="row<%= iRow %>PopoverAdd"
+                      class="dropdown-item js-group-action"
+                      data-toggle="popover"
+                      data-container="body"
+                      data-html="true"
+                      data-placement="auto"
+                      title="Add Members"
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs:267]: Nit:

```suggestion
                      title="Add members"
```

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs [hunk 5]
```diff
@@ -240,64 +248,55 @@
...
+                      id="row<%= iRow %>Popoverdeletemember"
+                      class="dropdown-item js-group-action"
+                      <% if (row.users.length === 0) { %>
+                      disabled
+                      <% } else { %> 
+                      data-toggle="popover"
+                      data-container="body"
+                      data-html="true"
+                      data-placement="auto"
+                      title="Remove Members"
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs:282]: Nit:

```suggestion
                      title="Remove members"
```

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs [hunk 6]
```diff
@@ -240,64 +248,55 @@
...
-
+                    </button>
+                    <button
+                      id="row<%= iRow %>Popoverdeletegroup"
+                      class="dropdown-item js-group-action"
+                      data-toggle="popover"
+                      data-container="body"
+                      data-html="true"
+                      data-placement="auto"
+                      title="Confirm delete this group"
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs:295]: Nit:
```suggestion
                      title="Delete group"
```

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs [hunk 7]
```diff
@@ -240,64 +248,55 @@
...
-                  </div>
-
-                  <%# Capture all clicks to dropdown items to prevent scrolling to the top of the page %>
-                  <div class="dropdown-menu" onclick="window.event.preventDefault();">
-
-                    <% if (authz_data.has_course_instance_permission_edit) { %>
-                    <a class="dropdown-item" onclick="$('#row<%= iRow %>PopoverAdd').popover('show')" href="#">
+                  <div class="dropdown-menu">
+                    <button
+                      id="row<%= iRow %>PopoverAdd"
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs:261]: I believe we no longer require IDs on all these buttons.

Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs:261]: Ah, right.

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs (hunk 3 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs]: Indeed. I have removed the second check in 4c8e77c9fec320280a6df812ab7c5d4f06265331. This still causes the dropdown menu to close when clicking inside the popover, but this behaviour was already happening in master. I added a comment about this as well.

Other review comments also addressed in the same commit.

Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs]: Adding the following snippet was sufficient to keep the dropdown open when clicking in the popover; I also added in your original bit to close the popover when the dropdown closes.

```js
      $('.js-group-action-dropdown').on('hide.bs.dropdown', (e) => {
        // If the click is inside a popover, don't hide the dropdown.
        if (e.clickEvent.target.closest('.popover')) {
          e.preventDefault();
          e.stopPropagation();
          return;
        }

        // Hide all popovers when the dropdown menu is closed.
        $('.js-group-action[data-toggle="popover"]').popover("hide");
      });
```

Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.ejs]: This worked really well, it even keeps the dropdown open if I hit "Cancel" in the popover (which is interesting).

**Pull request merged**